lib/spack/spack/database.py

```diff
@@ -638,14 +638,18 @@ def _read_spec_from_dict(self, hash_key, installs):
         Does not do any locking.
         """
         spec_dict = installs[hash_key]['spec']
-
+        spec_path = installs[hash_key].get('path', None)
         # Install records don't include hash with spec, so we add it in here
         # to ensure it is read properly.
         for name in spec_dict:
             spec_dict[name]['hash'] = hash_key
 
         # Build spec from dict first.
         spec = spack.spec.Spec.from_node_dict(spec_dict)
+
+        # Make sure prefix doesn't change
+        if spec_path and spec_path != 'None':
+            spec.prefix = spec_path
```

Comment:
In which cases would the string `None` be returned?

Reply:
An older database format had `None` for the path for externals